### ISSUE TITLE
feat (pop api): add burn and mint to local fungibles use case

### ISSUE THREAD

**Issue by Daanvdplas** [r0gue-io/pop-node#98]
Based on the `Mintable` and `Burnable` standards.

standards:
- [mint](https://docs.openzeppelin.com/contracts/4.x/api/token/erc20#ERC20-_mint-address-uint256-)
- [burn](https://docs.openzeppelin.com/contracts/4.x/api/token/erc20#ERC20Burnable-burn-uint256-)

As you can see they also emit the `Transfer` event

**Comment by chungquantin** [comment]
- Reference for native token: https://substrate.stackexchange.com/questions/1071/is-there-any-way-to-mint-new-native-tokens-after-deploying-the-substrate-chain
- With non-native fungible asset, we can basically call the dispatchable functions directly
```rust
#[pallet::call_index(6)]
pub fn mint(
	origin: OriginFor<T>,
	id: T::AssetIdParameter,
	beneficiary: AccountIdLookupOf<T>,
	#[pallet::compact] amount: T::Balance,
) -> DispatchResult

#[pallet::call_index(7)]
pub fn burn(
	origin: OriginFor<T>,
	id: T::AssetIdParameter,
	who: AccountIdLookupOf<T>,
	#[pallet::compact] amount: T::Balance,
) -> DispatchResult
```

**Comment by Daanvdplas** [comment]
For the native token we won't be able to mint since we are using the DOT token. Only the polkadot protocol can mint tokens through inflation.

**Comment by Daanvdplas** [comment]
This will still require the root origin most of the time because allowing anyone to mint the native token of the chain seems like a unlikely configuration (difficult to create value for your token if everyone can create them our of thin air). A chain using a contract to mint its native token could be a use case but TMO not reason enough to include it now. In general, the `force_` dispatchables have also not been added to the fungibles use case for the same reason. Depending on use cases and developer demand we can always implement them.

**Comment by Daanvdplas** [comment]
Closed with #150
